bin/node/src/config.rs

```diff
@@ -1,35 +1,71 @@
+use std::path::Path;
+
 use miden_node_block_producer::config::BlockProducerConfig;
 use miden_node_rpc::config::RpcConfig;
 use miden_node_store::config::StoreConfig;
+use miden_node_utils::config::Endpoint;
 use serde::{Deserialize, Serialize};
 
 /// Node top-level configuration.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+#[derive(Clone, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
 pub struct NodeConfig {
-    pub block_producer: Option<BlockProducerConfig>,
-    pub rpc: Option<RpcConfig>,
-    pub store: Option<StoreConfig>,
+    pub block_producer: BlockProducerConfig,
+    pub rpc: RpcConfig,
+    pub store: StoreConfig,
 }
 
-impl Default for NodeConfig {
-    fn default() -> Self {
-        Self {
-            block_producer: Some(Default::default()),
-            rpc: Some(Default::default()),
-            store: Some(Default::default()),
+impl NodeConfig {
+    pub fn load(config_file: impl AsRef<Path>) -> figment::Result<Self> {
+        /// A helper struct to deserialize a subset of [BlockProducerConfig] options.
+        #[derive(Deserialize)]
+        struct UserBlockProducerConfig {
+            endpoint: Endpoint,
+            verify_tx_proofs: bool,
+        }
+
+        /// A helper struct to deserialize a subset of [RpcConfig] options.
+        #[derive(Deserialize)]
+        struct UserRpcConfig {
+            endpoint: Endpoint,
+        }
+
+        /// A normalized version of [NodeConfig] i.e. with all redundant options
+        /// removed. This helps prevent user configuration issues.
+        #[derive(Deserialize)]
+        struct UserConfig {
+            block_producer: UserBlockProducerConfig,
+            rpc: UserRpcConfig,
+            store: StoreConfig,
         }
+
+        let config =
+            miden_node_utils::config::load_config(config_file.as_ref()).extract::<UserConfig>()?;
```

Comment:
Since figment is now a dependency - should we just call it directly (i.e., `Figment::from(Toml::file(config_file))`). In fact, i'm not sure our `load_config()` function is all that useful any more - so, maybe it can be removed altogether and we just use `NodeConfig::load()` everywhere?

Reply:
I thought maybe the `load_config` was isolated so that there would be a single place where we manipulate the configuration provider e.g. if we wanted to add env flag support then this would be the single place to add it.

I think probably its a good idea to encapsulate this at least somewhat if we have many config structs. I'll see how things look after fixing the component configurations.